File: python/second/concatenation_of_array.py
class Solution(object):
    # O(n) time | O(n) space
    def getConcatenation(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = [0] * 2 * len(nums)

        for i in range(2 * len(nums)):
            res[i] = nums[i % len(nums)]
        return res


solution = Solution()
nums = [1,2,1]
print(solution.getConcatenation(nums))

File: src/main/java/october_2025/ConcatenationOfArray.java
package october_2025;

public class ConcatenationOfArray {

    public static void main(String[] args) {
        int[] nums = {1,2,1};

        ConcatenationOfArray concatenationOfArray = new ConcatenationOfArray();
        int[] result = concatenationOfArray.getConcatenation(nums);
        System.out.println(result);

    }

    // O(n) time | O(n) space
    public int[] getConcatenation(int[] nums) {
        int[] result = new int[2 * nums.length];

        for (int i = 0; i < result.length; i++) {
            result[i] = nums[i % nums.length];
        }

        return result;
    }

}
